plugins/Goals/Goals.php

```diff
@@ -292,6 +309,31 @@ private static function getAllReportsWithGoalMetrics()
             }
         }
 
+        $reportsWithGoals[] = array('category' => 'Pages',
+            'name'     => Piwik::translate('General_Pages'),
+            'module'   => 'Goals',
+            'action'   => 'getPagesUrl',
+            'viewDataTable' => 'tableGoalsPages',
+        );
+        $reportsWithGoals[] = array('category' => 'Pages',
+            'name'     => Piwik::translate('Goals_EntryPages'),
+            'module'   => 'Goals',
+            'action'   => 'getPagesEntry',
+            'viewDataTable' => 'tableGoalsEntryPages',
+        );
+        $reportsWithGoals[] = array('category' => 'Pages',
+            'name'     => Piwik::translate('Goals_PageTitles'),
+            'module'   => 'Goals',
+            'action'   => 'getPagesTitles',
+            'viewDataTable' => 'tableGoalsPages',
+        );
+        $reportsWithGoals[] = array('category' => 'Pages',
+            'name'     => Piwik::translate('Goals_EntryPagesTitles'),
+            'module'   => 'Goals',
+            'action'   => 'getPagesEntryTitles',
+            'viewDataTable' => 'tableGoalsEntryPages',
+            'noMenu'   => true
+        );
```

Comment:
The special handling of the pages reports menu was done partly as wasn't aware of getDefaultTypeViewDataTable() and partly because the original issue UI mock up specifically excludes the EntryPagesTitles report from the side menu in favor of having it as a related report. I've replaced the noMenu flag with a specifically excluding the report by name when building the menu in `buildGoalByDimensionView`, adjusted the reports to use getDefaultTypeViewDataTable() and removed the special handling.